pygmt/src/config.py

```diff
@@ -55,7 +55,7 @@ def __init__(self, **kwargs):
                     self.old_defaults[key] = lib.get_default(key)
 
         # call gmt set to change GMT defaults
-        arg_str = " ".join([f"{key}={value}" for key, value in kwargs.items()])
+        arg_str = " ".join([f'{key}="{value}"' for key, value in kwargs.items()])
```

Comment:
> xref: https://forum.generic-mapping-tools.org/t/errors-in-pygmt-when-there-are-spaces-in-string-arguments/2494/3

Ok, I've fixed the `FORMAT_DATE_MAP="o dd"` issue reported in the forum by wrapping the arguments in double quotes here in commit c29e63229f6c46aeb47cfe068b9f6c959728f64a. Didn't use `\040` because `o\040dd` doesn't work. The problem with this solution is that the workaround suggested in the forum (use `FORMAT_DATE_MAP='"o dd"'`) will break in PyGMT v0.6.0, but I think that's acceptable since that workaround isn't intuitive anyway.